### PR TITLE
Fix generateShortOrCompleteClassname

### PR DESCRIPTION
**Pull request by qRoC**

### CODE REVIEW

**Comment by veewee** [comment]
Looks good.
Can you also add a test for this case so that I don't make this mistake again? Thanks!
I hope this one gets patched fast :)

**Comment by Ocramius** [comment]
As mentioned by @veewee, a test case is strictly required for fixes to be merged.

**Comment by aschwin** [comment]
Yes! Nice to see it fixed so quick!

**Comment by weierophinney** [comment]
@qRoC I really want to get this in, but I *really* need a test case, for a couple of reasons:

- *I* don't know how to reproduce the issue.
- *The project* needs to ensure it doesn't break in the future.

Any chance you can get a test in ASAP so I can do a 3.0.4 release, please?

**Comment by qRoC** [comment]
yes

File: test/Generator/ClassGeneratorTest.php [hunk 1]
```diff
@@ -1105,6 +1105,17 @@ public function testCorrectExtendNames()
...
+        $classGenerator = new ClassGenerator();
+        $classGenerator->setName('ClassName');
+        $classGenerator->setNamespaceName('SomeNamespace');
+        $classGenerator->setExtendedClass('\DateTime');
+        $this->assertContains('class ClassName extends \DateTime', $classGenerator->generate());
+
+        $classGenerator = new ClassGenerator();
+        $classGenerator->setName('ClassName');
+        $classGenerator->setExtendedClass('\DateTime');
+        $this->assertContains('class ClassName extends DateTime', $classGenerator->generate());
```
Comment (weierophinney) [inline test/Generator/ClassGeneratorTest.php]: Please create new test methods for these two new cases, using a name descriptive of the behavior being tested, and annotate them with `@group 75`.

Thanks!

**Pull request merged**